doc/contributing.rst

```diff
@@ -21,6 +21,37 @@ of the users who use the package.
 page as the maintainers about changes or enhancements before too much
 coding is done saves everyone time and effort!
 
+Installing developer version
+----------------------------
+
+The standard installation procedure can be found on
+:ref:`install_python_and_mne_python` but if you want to install a
+developer version directly from the source code in order to contribute,
+you can follow those steps:
+
+  .. code-block:: console
+
+     $ git clone git@github.com:mne-tools/mne-python.git
+     $ cd mne-python
+     $ conda env create -f environment.yml
+     $ conda activate mne
+     $ pip install -e .
+
+As optional steps, you can also add:
```

Comment:
why would I want to do this? Build the doc? It's what you have in mind?

Reply:
Quoting from [Pip documentation](https://pip.pypa.io/en/stable/reference/pip_install/?highlight=--editable), using `--editable` or `-e`, `pip` has the capability to install directly from a version control repository or a local project path.
The goal here is to install from the source code cloned from `git`.

Reply:
Oh, if you talk about the optional steps, yes it's for building documentation.

Reply:
can you add a section then on what needs to be installed in order to build the documentation? thanks